src/Chainweb/Mempool/RestAPI/Client.hs

```diff
@@ -217,7 +243,29 @@ subscribeClient v c = runIdentity $ do
 
 ------------------------------------------------------------------------------
 #if MIN_VERSION_servant(0,15,0)
-#error TODO: need to support servant >= 0.15
+
+-- TODO: the code in this module could be simplfied by replacing the use of
+-- io-streams with servant's build-in SourceIO stream type.
+--
+instance Show a => FromSourceIO a (Streams.InputStream a) where
+    fromSourceIO (SourceT src) = unsafePerformIO $ src $ \step ->
```

Comment:
😬

Reply:
This also was in the original code for servant 0.14. 

My understanding is that it is due to the design of io-streams to implement the underlying stream loop with mutable state (`IORef`). So, unlike with other streaming frameworks, initializing a stream requires an IO action that isn't an effect in the stream itself.

Reply:
Ahh yes.  I missed that.  Carry on.  👍